datasource/observer/clients/restobserverclient.js

```diff
@@ -40,74 +39,101 @@ function setupEndpoints ()
 			}
 		};
 		this.notify (notification);
-
 		res.sendStatus(200);
 	}
 
-	this._webServer.put ('/repository/:repoName',
+	this._app.post ('/repository/:repoName',
 		(req, res) => repoCb (req, res, Notification.action.CREATE));
 
-	this._webServer.delete ('/repository/:repoName',
+	this._app.delete ('/repository/:repoName',
 		(req, res) => repoCb (req, res, Notification.action.DELETE));
 
-	this._webServer.put ('/repository/:repoName/entity',
+	this._app.post ('/repository/:repoName/entity',
 		(req, res) => entitiesCb (req, res, Notification.action.CREATE));
 
-	this._webServer.post ('/repository/:repoName/entity',
+	this._app.put ('/repository/:repoName/entity',
 		(req, res) => entitiesCb (req, res, Notification.action.UPDATE));
 
-	this._webServer.delete ('/repository/:repoName/entity',
+	this._app.delete ('/repository/:repoName/entity',
 		(req, res) => entitiesCb (req, res, Notification.action.DELETE));
 }
 
 class RestObserverClient extends ObserverClient
 {
-	constructor (info, options)
+	/**
+	 * 
+	 * @param {Object} info observer info from hkbase
+	 * @param {Object} options observer initialization options
+	 * @param {string} options.baseUrl base URL of hkbase to be used for registering observer
+	 * @param {number} options.port port te be used when instantiating express server for receiving callback requests
+	 * @param {string} options.address address to be used when instantiating express server for receiving callback requests
+	 * @param {Object} hkbaseOptions options to be used when communicating with hkbase
+	 * @param {Object} observerServiceParams observer service parameters (if using specialized observer)
+	 */
+	constructor (info, options, hkbaseOptions, observerServiceParams)
 	{
-		super ();
+		super (hkbaseOptions, observerServiceParams);
 		this._baseUrl   = options.baseUrl;
-		this._webServer = express ();
+		this._app = express ();
 		this._port      = options.port || 0;
 		this._address   = options.address || DEFAULT_ADDR;
+		this._observerId = null;
+		this._listeningPath = null;
+		this._server = null;
+		
 
 		if (!this._baseUrl.endsWith('/'))
 		{
 			this._baseUrl = `${this._baseUrl}/`;
 		}
 
-		this._webServer.use (bodyParser.json());
+		this._app.use (bodyParser.json());
 	}
 
 	static getType ()
 	{
 		return 'rest';
 	}
 
-	init ()
+	async init ()
 	{
+		console.info(`initializing REST observer client`);
 		return new Promise ((resolve, reject) =>
 			{
 				setupEndpoints.call (this);
```

Comment:
Is there any benefit for having this method on its own and not a member of the class?

Reply:
Not really, but I didn't see the need of changing this.